dali/dfuplus/dfuplus.cpp

```diff
@@ -1448,7 +1452,262 @@ int CDfuPlusHelper::abort()
+                    info("%s", propVal);
+                }
+                info("\n");
+            }
+        }
+        break;
+
+    case json:
+        {
+            const char * level1 = "   ";        // 3 space 'tab'
+            const char * level2 = "      ";
+            const char * level3 = "         ";
+            //                     123456789
+
+            unsigned index = 1;
+
+            info("{\n");       // Level 0: Start History object
+            StringBuffer temp;
+            history->getName(temp);
+            info("%s\"%s\": [", level1, temp.str());   // Begin History object's array of historical records
+            Owned<IPropertyTreeIterator> historyIter = history->getElements("*");
+            ForEach(*historyIter)
+            {
+                if (index++ != 1)
+                    info(",");
+
+                info("\n%s{\n", level2);    // Historical record object begin
+
+                bool first = true;
+                Owned<IAttributeIterator> attribIter = historyIter->query().getAttributes(sorted);
+                ForEach(*attribIter)
+                {
+                    if (!first)
+                        info(",\n");
+                    else
+                        first = false;
+
+                    const char *propName = attribIter->queryName();
+                    if (*propName == '@')
+                        propName++;
+                    const char *propVal =  attribIter->queryValue();
+                    info("%s\"%s\" : \"%s\"",level3, propName, propVal);    // Historical record item object
+                }
+                info("\n%s}", level2);   // Historical record object end
+            }
+            info("\n%s]\n", level1);   // History object array end
+
+            info("}\n");       // Level 0 : History object end
+        }
+        break;
+    }
+
+    return 0;
+}
+
+int CDfuPlusHelper::erasehistory()
+{
+    const char* lfn = globals->queryProp("lfn");
+    if(lfn == NULL || *lfn == '\0')
+        throw MakeStringException(-1, "srcname not specified");
+
+    bool backup = globals->getPropBool("backup", true);
+    const char* dstxml = globals->queryProp("dstxml");
+    if(backup && (dstxml == NULL || *dstxml == '\0'))
+        throw MakeStringException(-1, "dstxml not specified");
+
+    progress("\nErase history of '%s' with%s backup.\n", lfn, (backup ? "" : "out"));
+
+    Owned<IClientEraseHistoryRequest> req = dfuclient->createEraseHistoryRequest();
+    req->setName(lfn);
+
+    Owned<IClientEraseHistoryResponse> resp = dfuclient->EraseHistory(req);
+
+    const IMultiException* excep = &resp->getExceptions();
+    if(excep != NULL && excep->ordinality() > 0)
+    {
+        StringBuffer errmsg;
+        excep->errorMessage(errmsg);
+        error("%s\n", errmsg.str());
+        return -1;
+    }
+
+    if (backup)
+    {
+        MemoryBuffer tmp;
+        tmp.append(resp->getXmlmap());
```

Comment:
See prev. comment re. this, should cast rather than copy and comment to it's constantness semantics